### ISSUE TITLE
v2 Istio fails to completely install/upgrade when Ingress Gateway is not installed

### ISSUE THREAD

**Issue by bmdepesa** [rancher/rancher#29383]
rancher/rancher:v2.5.0-rc8

* Deploy Rancher
* Deploy a cluster
* In Cluster Explorer, deploy Istio app, while deselecting `Ingress Gateway` in components during installation
![Screen Shot 2020-10-05 at 12 25 04 PM](https://user-images.githubusercontent.com/21316915/95122754-d2ee5e00-0705-11eb-949f-8c97b732725d.png)
* Istio fails to install with an error:
```
Checked 0 Istio Deployments
No Istio installation found
Error: no Istio installation found
```

* The same occurs if you try to upgrade an installation to remove ingress gateway

Seems to be related to this issue: https://github.com/istio/istio/issues/24374 where verify install fails when the default ingress gateway is not present

* Tested with disabling other components and `istio-ingress-gateway` seems to be the common link

**Comment by bmdepesa** [comment]
rancher/rancher:master-head 0390ff5

* Deploy Rancher
* Create a cluster
* Install Istio 1.7.3 without Ingress Gateway selected
* Install is successful
* Upgrade to add/remove Ingress Gateway successfully

**Comment by murphye** [comment]
This is still a problem in 1.8.3. Recommend to reopen.

**Comment by cbron** [comment]
@murphye you mean in upstream Istio or via Rancher ? We haven't released 1.8.3 yet.
